compiler/resolution/implementForallIntents.cpp

```diff
@@ -887,12 +887,23 @@ static VarSymbol* localizeYieldForExtendLeader(Expr* origRetExpr, Expr* ref) {
   Symbol* origRetSym = orse->var;
   for (Expr* curr = ref->prev; curr; curr = curr->prev)
     if (CallExpr* call = toCallExpr(curr))
-      if (call->isPrimitive(PRIM_MOVE))
+      if (call->isPrimitive(PRIM_MOVE) ||
+          call->isNamed("="))
         if (SymExpr* dest = toSymExpr(call->get(1)))
           if (dest->var == origRetSym) {
             VarSymbol* newOrigRet = newTemp("localRet", origRetSym->type);
-            curr->insertBefore(new DefExpr(newOrigRet));
+            call->insertBefore(new DefExpr(newOrigRet));
             dest->var = newOrigRet;
+            if (call->isNamed("=")) {
+              // We are "initializing" localRet, not "assigning" to it.
```

Comment:
Assignment might be "correct" in the case where the return value variable has a declared type and does not support "noinit".  Which is to say we cannot assume that we are always initializing the RVV in this context.  (At least, not without more experimentation.
Just the same, this change is acceptable, because the case that I pointed out will give rise to memory leaks if it exists -- whereas this PR fixes a correctness problem.  We will revisit this code either when chasing down memory leaks or when updating the internal representation so that returns and yields are always implemented as initialization (never assignment).
Summary: No change required; I am just recording my current thoughts on the subject.